panic: Unable to synchronize wallet to chain (during restore)

### Background

When restoring an LND wallet with a seed, LND panics on sync with the following error:

```
panic: Unable to synchronize wallet to chain: unable to perform wallet recovery: failed to store sync information
```

Here is the full error message:

```
2019-06-19 15:11:46.502 [INF] LNWL: RECOVERY MODE ENABLED -- rescanning for used addresses with recovery_window=2500
2019-06-19 15:11:46.505 [INF] LNWL: Seed birthday surpassed, starting recovery of wallet from height=1544150 hash=00000000000d71fb89ae8a84ad174ce9fa62b63ca1a040ddeaffa901ddfaf07b with recovery-window=2500
panic: Unable to synchronize wallet to chain: unable to perform wallet recovery: failed to store sync information 00000000000d71fb89ae8a84ad174ce9fa62b63ca1a040ddeaffa901ddfaf07b: failed to fetch block hash for height 1544149: block not found

goroutine 133705 [running]:
github.com/btcsuite/btcwallet/wallet.(*Wallet).handleChainNotifications(0xc0001e8a00)
	/Users/conner/go/pkg/mod/github.com/btcsuite/btcwallet@v0.0.0-20190614043544-a335c566148c/wallet/chainntfns.go:117 +0x149d
created by github.com/btcsuite/btcwallet/wallet.(*Wallet).SynchronizeRPC
	/Users/conner/go/pkg/mod/github.com/btcsuite/btcwallet@v0.0.0-20190614043544-a335c566148c/wallet/wallet.go:191 +0x153
```

After doing a quick GH investigation, my hunch would be this `btcwallet` PR:

https://github.com/btcsuite/btcwallet/pull/618
https://github.com/btcsuite/btcwallet/blob/a335c566148cfbe4a90e191564f2146525fe5861/wallet/chainntfns.go#L117

I've been able to reproduce by simply restoring with seed, but only have tested while using `neutrino` as the backend. 

### Your environment

* version of `lnd`
0.7.0-rc-1

* version of `btcd`, `bitcoind`, or other backend
Neutrino

### Steps to reproduce

- Create a new LND wallet with seed
- Attempt to restore with seed

--- a/go.mod
+++ b/go.mod
@@ -29,7 +29,7 @@ require (
 	github.com/juju/utils v0.0.0-20180820210520-bf9cc5bdd62d // indirect
 	github.com/juju/version v0.0.0-20180108022336-b64dbd566305 // indirect
 	github.com/kkdai/bstream v0.0.0-20181106074824-b3251f7901ec
-	github.com/lightninglabs/neutrino v0.0.0-20190606084233-98b149cd0468
+	github.com/lightninglabs/neutrino v0.0.0-20190620074315-32af2fbd0d2b
 	github.com/lightningnetwork/lightning-onion v0.0.0-20190430041606-751fb4dd8b72
 	github.com/lightningnetwork/lnd/queue v1.0.1
 	github.com/lightningnetwork/lnd/ticker v1.0.0

--- a/go.mod
+++ b/go.mod
@@ -8,7 +8,7 @@ require (
 	github.com/btcsuite/btcd v0.0.0-20190614013741-962a206e94e9
 	github.com/btcsuite/btclog v0.0.0-20170628155309-84c8d2346e9f
 	github.com/btcsuite/btcutil v0.0.0-20190425235716-9e5f4b9a998d
-	github.com/btcsuite/btcwallet v0.0.0-20190619005538-de02d6fdfb23
+	github.com/btcsuite/btcwallet v0.0.0-20190620233257-46c0cf2a3f3a
 	github.com/btcsuite/fastsha256 v0.0.0-20160815193821-637e65642941
 	github.com/coreos/bbolt v1.3.2
 	github.com/davecgh/go-spew v1.1.1

